lib/pundit.rb

```diff
@@ -80,10 +80,13 @@ def authorize(user, record, query, policy_class: nil)
     # @raise [InvalidConstructorError] if the policy constructor called incorrectly
     # @return [Scope{#resolve}, nil] instance of scope class which can resolve to a scope
     def policy_scope(user, scope)
-      policy_scope = PolicyFinder.new(scope).scope
+      begin
+        policy_scope = PolicyFinder.new(scope).scope
+      rescue ArgumentError
+        raise InvalidConstructorError, "Invalid #<#{policy_scope}> constructor is called"
+      end
+
       policy_scope.new(user, pundit_model(scope)).resolve if policy_scope
```

Comment:
As we discussed in the PR #548 the call we want to rescue errors from is `policy_scope.new(user, pundit_model(scope))`. So you have to move that into the rescue block and only run the `resolve` call outside of it.

Reply:
I pushed some code with what I said. Let me know if it's good for you.